shelley/chain-and-ledger/executable-spec/src/STS/NewEpoch.hs

```diff
@@ -98,7 +98,9 @@ newEpochTransition = do
           sd =
             aggregatePlus $
               catMaybes
-                [ (,fromIntegral c / fromIntegral total) <$> Map.lookup hk delegs
+                [ (,fromIntegral c / fromIntegral (if total == 0 then 1 else total)) <$>
+                  Map.lookup hk delegs -- TODO mgudemann total could be zero (in
+                                       -- particular when shrinking)
```

Comment:
good catch! I guess the justification for using `1` in the event that there is not total stake is that then any given pool's stake is zero, so we compute `0/1` instead of `0/0`.

Reply:
exactly!